src/main/java/org/jabref/gui/util/BackgroundTask.java

```diff
@@ -49,6 +51,15 @@ protected V call() throws Exception {
         };
     }
 
+    public static <V> BackgroundTask<V> wrap(Supplier<V> supplier) {
```

Comment:
What's the reason for this addition? Callable and Supplier are essentially equivalent, just that callable allows checked exceptions (which is what we want).

Reply:
When working with `CompletableFuture` it's always handy to have a method available that accepts the Supplier interface. The `CompletableFuture` only accepts either `Runnable` or `Supplier` types when submitting a new task. Check out the static methods [here](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html)

Reply:
Ok, but there shouldn't ever be a need to use these static methods directly as everything should be wrapped by a background task. Please remove this for the moment. In case, we need it later it's easy to add.